packages/next/src/server/app-render/app-render.tsx

```diff
@@ -1513,6 +1513,23 @@ export const renderToHTMLOrFlight: AppPageRender = (
   )
 }
 
+const addHeaderToResponse = (
+  res: BaseNextResponse,
+  key: string,
+  value: string | string[]
+) => {
+  if (key === 'link' && res.getHeader(key)) {
```

Comment:
I've currently applied the behavior for merging header values only for `link`. That's the only one I've encountered the issue with so far and it also seems to be [the only one that React sets](https://github.com/facebook/react/blob/5b0ef217ef32333a8e56f39be04327c89efa346f/packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js#L6292) currently.